src/diff/props.js

```diff
@@ -103,49 +97,39 @@ export function setProperty(dom, name, value, oldValue, isSvg) {
 		} else {
 			dom.removeEventListener(name, proxy, useCapture);
 		}
-	} else if (
-		name !== 'list' &&
-		name !== 'tagName' &&
-		// HTMLButtonElement.form and HTMLInputElement.form are read-only but can be set using
-		// setAttribute
-		name !== 'form' &&
-		name !== 'type' &&
-		name !== 'size' &&
-		name !== 'download' &&
-		name !== 'href' &&
-		name !== 'contentEditable' &&
-		!isSvg &&
-		name in dom
-	) {
-		dom[name] = value == null ? '' : value;
-	} else if (typeof value != 'function' && name !== 'dangerouslySetInnerHTML') {
-		if (name !== (name = name.replace(/xlink:?/, ''))) {
-			if (value == null || value === false) {
-				dom.removeAttributeNS(
-					'http://www.w3.org/1999/xlink',
-					name.toLowerCase()
-				);
-			} else {
-				dom.setAttributeNS(
-					'http://www.w3.org/1999/xlink',
-					name.toLowerCase(),
-					value
-				);
-			}
+	} else if (name !== 'dangerouslySetInnerHTML') {
+		if (isSvg) {
+			// Normalize incorrect prop usage for SVG:
+			// - xlink:href / xlinkHref --> href (xlink:href was removed from SVG and isn't needed)
+			// - className --> class
+			name = name.replace(/xlink[H:h]/, 'h').replace(/Name$/, '');
```

Comment:
This might break the [`attributeName` SVG attribute](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/attributeName)

Reply:
I think this works. It adds a couple bytes but on my machine they are offset by the change I suggested in my other comment below.

```suggestion
			name = name.replace(/xlink[H:h]/, 'h').replace(/sName$/, 's');
```

Reply:
Ah good catch, I assumed I was safe because this was SVG land, but that's actually where casing goes wild haha.